src/main/scala/is/hail/sparkextras/OrderedRDD2.scala

```diff
@@ -14,16 +14,22 @@ class OrderedDependency(left: OrderedRVD, right: OrderedRVD) extends NarrowDepen
 
 object OrderedDependency {
   def getDependencies(p1: OrderedRVDPartitioner, p2: OrderedRVDPartitioner)(partitionId: Int): Range = {
-    val lastPartition = if (partitionId == p1.rangeBounds.length)
-      p2.numPartitions - 1
-    else
-      p2.getPartitionPK(p1.rangeBounds(partitionId))
+    val partBounds = p1.rangeBounds(partitionId).asInstanceOf[Interval]
 
-    if (partitionId == 0)
-      0 to lastPartition
+    if (!p2.rangeTree.overlaps(p2.pkType.ordering, partBounds))
+      0 to 0
```

Comment:
Not empty?

Reply:
oops, yes. should be 0 until 0. I don't know how else to create an empty Range, though.

Reply:
Change the return type to Seq[Int].